backend/python/tapa/safety_check.py

```diff
@@ -0,0 +1,21 @@
+import logging
+from typing import List
+
+from .task import Task
+
+DISABLED_MMAP_NAME_LIST = {"in", "out", "input", "output", "reg", "wire", "begin", "end"}
+
+_logger = logging.getLogger().getChild(__name__)
+
+
+def check_mmap_arg_name(task_list: List[Task]) -> None:
+  """
+  mmap arguments cannot be named "in" or "out". Otherwise HLS will have inconsistent naming
+  convention in the generated AXI interface
+  """
+  for task in task_list:
+    if task.is_upper:
+      for port_name in task.ports.keys():
+        if port_name in DISABLED_MMAP_NAME_LIST:
+          _logger.error(f'Task arguments cannot be among the reserved keywords: {DISABLED_MMAP_NAME_LIST}')
```

Comment:
For logging messages we usually use % style strings to enable lazy evaluation. Doesn't make a difference here, but helpful for debugging messages.